pyani_plus/private_cli.py

```diff
@@ -0,0 +1,379 @@
+    )
+
+    return 0
+
+
+@app.command()
+def log_comparison(  # noqa: PLR0913
+    database: Annotated[str, typer.Option(help="Path to pyANI-plus SQLite3 database")],
+    # These are for the comparison table
+    query_fasta: Annotated[Path, typer.Option(help="Path to query FASTA file")],
+    subject_fasta: Annotated[Path, typer.Option(help="Path to subject FASTA file")],
+    identity: Annotated[
+        float, typer.Option(help="Percent identity (float from 0 to 1)")
+    ],
+    aln_length: Annotated[int, typer.Option(help="Alignment length")],
+    # These are all for the configuration table:
+    method: Annotated[str, typer.Option(help="Comparison method")],
+    program: Annotated[str, typer.Option(help="Comparison program name")],
+    version: Annotated[str, typer.Option(help="Comparison program version")],
+    fragsize: Annotated[
+        int | None, typer.Option(help="Comparison method fragment size")
+    ] = None,
+    maxmatch: Annotated[
+        bool | None, typer.Option(help="Comparison method max-match")
+    ] = None,
+    kmersize: Annotated[
+        int | None, typer.Option(help="Comparison method k-mer size")
+    ] = None,
+    minmatch: Annotated[
+        float | None, typer.Option(help="Comparison method min-match")
+    ] = None,
+    # Optional comparison table entries
+    sim_errors: Annotated[int | None, typer.Option(help="Alignment length")] = None,
+    cov_query: Annotated[float | None, typer.Option(help="Alignment length")] = None,
+    cov_subject: Annotated[float | None, typer.Option(help="Alignment length")] = None,
+    create_db: Annotated[  # noqa: FBT002
+        bool, typer.Option(help="Create database if does not exist")
+    ] = False,
+) -> int:
+    """Log a single pyANI-plus pairwise comparison to the database."""
+    if database != ":memory:" and not create_db and not Path(database).is_file():
+        msg = f"ERROR: Database {database} does not exist, but not using --create-db"
+        sys.exit(msg)
+
+    print(f"Logging to {database}")  # noqa: T201
+    session = db_orm.connect_to_db(database)
+
+    config = db_orm.add_configuration(
+        session, method, program, version, fragsize, maxmatch, kmersize, minmatch
+    )
+    if config.configuration_id is None:
+        sys.exit("Error with configuration table?")
+
+    query_md5 = file_md5sum(query_fasta)
+    db_orm.add_genome(session, query_fasta, query_md5)
+
+    subject_md5 = file_md5sum(subject_fasta)
+    db_orm.add_genome(session, subject_fasta, subject_md5)
+
+    db_orm.add_comparison(
+        session,
+        configuration_id=config.configuration_id,
+        query_hash=query_md5,
+        subject_hash=subject_md5,
+        identity=identity,
+        aln_length=aln_length,
+        sim_errors=sim_errors,
+        cov_query=cov_query,
+        cov_subject=cov_subject,
+    )
+
+    session.commit()
+    return 0
+
+
+def parse_fastani_file(filename: Path) -> tuple[Path, Path, float, int, int]:
```

Comment:
We can move ``parse_fastani_file`` (and perhaps ``log_fastani``) into a methods module later (I didn't want to conflict with the ongoing ANIm and dnadiff work).